### PR TITLE
Middleware API to submit support tickets

### PR DESCRIPTION
**Pull request by william-gr**
Ticket: #24988

### CODE REVIEW

**Comment by araujobsd** [comment]
retest this please

File: src/middlewared/middlewared/plugins/support.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                'https://%s/%s/api/v1.0/categories' % (ADDRESS, sw_name),
+                data=json.dumps({
+                    'user': username,
+                    'password': password,
+                }),
+                headers={'Content-Type': 'application/json'},
+                timeout=10,
+            )
+            data = r.json()
+        except simplejson.JSONDecodeError as e:
```
Comment (araujobsd) [inline src/middlewared/middlewared/plugins/support.py]: Could remove variable e if will not be used.

File: src/middlewared/middlewared/plugins/support.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+            self.logger.debug("Failed to decode ticket attachment response: %s", r.text)
```
Comment (araujobsd) [inline src/middlewared/middlewared/plugins/support.py]: Could use format.

File: src/middlewared/middlewared/plugins/support.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            raise CallError('Invalid proxy server response', errno.EBADMSG)
+        except requests.ConnectionError as e:
+            raise CallError(f'Connection error {e}', errno.EBADF)
+        except requests.Timeout as e:
```
Comment (araujobsd) [inline src/middlewared/middlewared/plugins/support.py]: Same here.

**Comment by kmoore134** [comment]
retest this please

**Comment by william-gr** [comment]
@araujobsd changes committed

**Pull request merged**